### PR TITLE
75 restructuring data for endpoints

### PR DESCRIPTION
**Pull request by jared23m**
I created a getUserPageById and trimmed all of the data down within the queries of the embedded functions. Overall the data being sent to the endpoint functions are much better formatted for front end use now, based on the wireframe we discussed.

I also trimmed the data down within the queries of the functions that make up getAllRecipesPage and getReviewedRecipesPage.

Finally I added getTagsByUser.

Ultimately I made a lot of extra work for myself, basing the original test functions too much off of Juicebox. Many of those test functions are useless now. I will continue to call them in the seed file though, and if we want to clean it all up later we can do that.

I think Larry and I together can easily get all of the endpoints fully working by Thursday. I imagine authentication is gonna be the tiny hurdle for the endpoints, and everything else will be relatively smooth sailing. We can probably pull off the MVP besides styling by next Tuesday.

### CODE REVIEW

File: BE/db/index.js [hunk 1]
```diff
@@ -196,7 +196,27 @@ async function getUserInfoWithPasswordById(userId) {
     throw error;
   }
 }
+ 
+async function getUserPageById(userId){
```
Comment (ljreaux) [inline BE/db/index.js:200]: /users/me endpoint

File: BE/db/index.js [hunk 2]
```diff
@@ -282,7 +302,11 @@ async function getRecipesByUser(userId) {
   //GET USER PAGE RECIPE BY ID
 async function getUserPageRecipeById(recipeId){
```
Comment (ljreaux) [inline BE/db/index.js:303]: /users/:userId/recipes endpoint

File: BE/db/index.js [hunk 3]
```diff
@@ -306,6 +327,41 @@ async function getUserPageRecipeById(recipeId){
 
 }
 
+  //GET OTHER PAGE RECIPE BY ID
+  async function getOtherPageRecipeById(recipeId){
```
Comment (ljreaux) [inline BE/db/index.js:331]: What is this used for?

Comment (jared23m) [reply BE/db/index.js:331]: That is the recipe list data that includes userInfo, for getAllRecipesPage and getReviewedRecipesPage

**Review by ljreaux — APPROVED** [review]

**Review by amead99 — APPROVED** [review]

**Review by BarryFullStack — APPROVED** [review]

**Pull request merged**